go-controller/pkg/ovndbmanager/ovndbmanager.go

```diff
@@ -0,0 +1,217 @@
+			klog.Infof("Previous stale member found: %s...kicking", member[1])
+			_, stderr, err = appCtl("cluster/kick", dbName, member[1])
+			if err != nil {
+				klog.Errorf("Error while kicking old Raft member: %s, for address: %s in db: %s,"+
+					"stderr: %v, error: %v", member[1], addr, db, stderr, err)
+			}
+		}
+	}
+}
+
+// ensureClusterRaftMembership ensures there are no unknown members in the current Raft cluster
+func ensureClusterRaftMembership(db string) {
+	var knownMembers, knownServers []string
+
+	var dbName string
+	var appCtl func(args ...string) (string, string, error)
+
+	if strings.Contains(db, "ovnnb") {
+		dbName = "OVN_Northbound"
+		appCtl = util.RunOVNNBAppCtl
+		knownMembers = strings.Split(config.OvnNorth.Address, ",")
+	} else {
+		dbName = "OVN_Southbound"
+		appCtl = util.RunOVNSBAppCtl
+		knownMembers = strings.Split(config.OvnSouth.Address, ",")
+	}
+	for _, knownMember := range knownMembers {
+		server := strings.Split(knownMember, ":")
+		if len(server) < 3 {
+			klog.Warningf("Failed to parse known member: %s", knownMember)
+			continue
+		}
+		knownServers = append(knownServers, server[1])
+	}
+	out, stderr, err := appCtl("cluster/status", dbName)
+	if err != nil {
+		klog.Warningf("Unable to get cluster status for: %s, stderr: %v, err: %v", db, stderr, err)
+		return
+	}
+	r, _ := regexp.Compile(`([a-z0-9]{4}) at ((ssl|tcp):\[?[a-z0-9.:]+\]?)`)
+	members := r.FindAllStringSubmatch(out, -1)
+	kickedMembersCount := 0
+	for _, member := range members {
+		if len(member) < 3 {
+			klog.Warningf("Unable to find parse member: %s", member)
+			return
+		}
+		matchedServer := strings.Split(member[2], ":")
+		if len(matchedServer) < 3 {
+			klog.Warningf("Unable to parse address portion of the member entry: %s", matchedServer)
+			return
+		}
+		memberFound := false
+		for _, knownServer := range knownServers {
+			if knownServer == matchedServer[1] {
+				memberFound = true
+				break
+			}
+		}
+		if !memberFound && (len(members)-kickedMembersCount) > 3 {
+			// unknown member and we have enough members its safe to kick the unknown address
+			klog.Infof("Unknown Raft member found: %s, %s...kicking", member[1], member[2])
+			_, stderr, err = appCtl("cluster/kick", dbName, member[1])
+			if err != nil {
+				// warn only: we might fail to kick since other nodes will also be trying to kick the member
+				klog.Warningf("Error while kicking old Raft member: %s, for address: %s in db: %s,"+
+					"stderr: %v, err: %v", member[1], member[2], db, stderr, err)
+				continue
+			}
+			kickedMembersCount = kickedMembersCount + 1
+		}
+	}
+}
+
+// ensureDBHealth ensures that if the db is clustered, it is healthy by calling the cluster-check command
+// if the clustered db is corrupt, the db will be deleted and kube-master pod needs to be started again.
+func ensureDBHealth(db string) {
+	stdout, stderr, err := util.RunOVSDBTool("check-cluster", db)
+	if err != nil {
```

Comment:
if no error, should we still check for stdout and stderr and warn or debug info?

```
If ovsdb-tool detects unusual but not necessarily incorrect content, it prints a warning or warnings on stdout.
```

Reply:
we can print the warnings - but that's the best we can do?